connect-inject/handler.go

```diff
@@ -296,14 +339,56 @@ func (h *Handler) Mutate(req *v1beta1.AdmissionRequest) *v1beta1.AdmissionRespon
 		resp.PatchType = &patchType
 	}
 
+	// Check and potentially create Consul resources. This is done after
+	// all patches are created to guarantee the mutation succeeded before
```

Comment:
technically this doesn't guarantee the mutation succeeded since we still need to pass it back kube right? So maybe add a little here about how we know there's no errors on our side

Reply:
Updated to be clearer that it's just our patch creation we're waiting on. If kube fails this afterwards, we will have already modified Consul. Not sure there's a way around that.